utils/scraper_processing_utils.R

```diff
@@ -328,10 +340,12 @@ get_persons <- function(json_res){
             max_str = c(max_str, curr_text)
         }
         else if(length(str_idx) > 1){
-            # there is an error, substring found twice
-            print("error in parsing the speakers")
+            # the substring found twice
+            # so this means if you are searching for John
+            # you could get back John Houghton or John Lee
+            # lets just ignore it
+            print("ambiguous speakers, (names are too similar)")
```

Comment:
I assume this is a genderize limitation, but could you input last name along with the first? Would that aid in disambiguating speakers at all?

Reply:
Hm, this is a good suggestion. I used to do this, but have started to reach my query/day limit. I moved to first name only so that I could easily cache the results and use them again later. Including last names would go beyond my daily query limit.

Is the worry here last name vs. first name mix-ups? I could additionally store the first + last name when I don't know the gender, then do a multi-pass gender prediction. It could go like this: 
- first pass: predict on assumed first name, 
- if no gender predicted run second pass on first + last name prediction

Would this address your concern?

Reply:
>Is the worry here last name vs. first name mix-ups? I could additionally store the first + last name when I don't know >the gender, then do a multi-pass gender prediction. It could go like this:
>
>    first pass: predict on assumed first name,
>    if no gender predicted run second pass on first + last name prediction
>
>Would this address your concern?

Oh I see. I was curious to know if adding last name changed/improved performance. If it reaches your daily limit feel free to ignore as I was just throwing an idea out there for genderize.

Reply:
aha, I see. Good check! Well when looking at the benchmark dataset, it actually looks like prediction accuracy went down significantly... F1 went from 0.703 to 0.615
Ok need to look into this a bit and find out why...

Reply:
phew, ok that took a while, but the fix is in commit 337d1ac
The reason why the prediction changed between the previous version and this one are for 3 reasons:

1. quote character bug: I use quotes as an ID to join against other tables. The bug was introduced when one table was read in to take into account quotation marks and the other was not. Now when these tables are read, all quote characters are stripped 
2. I now use generize.io for all predictions, I don't use anything predicted from coreNLP
3. I use the probability >=0.5 for male, previously the probability had to be > 0.6 for male, < 0.4 for female.

With each of these changes we now get better prediction metrics than before! The reason why we get more quotes to do the prediction on, is because if the quote character bug, which was present before, but not caught.

Previous gender prediction metrics:

![curr_gender_pred_metrics](https://user-images.githubusercontent.com/1885501/106534201-c958e500-64b0-11eb-8e40-c6d56d1af11a.png)

Current gender prediction metrics:

![prev_gender_pred_metrics](https://user-images.githubusercontent.com/1885501/106534203-c9f17b80-64b0-11eb-9503-958feb7c6ad8.png)